napari/layers/utils/text.py

```diff
@@ -60,44 +75,32 @@ class TextManager:
         Set to true of the text should be displayed.
     """
 
-    def __init__(
-        self,
-        text,
-        n_text,
-        properties={},
-        rotation=0,
-        translation=0,
-        anchor='center',
-        color='cyan',
-        size=12,
-        blending='translucent',
-        visible=True,
-    ):
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            text=Event,
-            rotation=Event,
-            translation=Event,
-            anchor=Event,
-            color=Event,
-            size=Event,
-            blending=Event,
-            visible=Event,
-        )
-
-        self.events.block_all()
-        self._rotation = rotation
-        self._anchor = Anchor(anchor)
-        self._translation = translation
-        self._color = transform_color(color)[0]
-        self._size = size
-        self._blending = self._check_blending_mode(blending)
-        self._visible = visible
-
-        self._set_text(text, n_text, properties)
-        self.events.unblock_all()
+    text: InitVar[Union[np.ndarray, str]]
+    n_text: InitVar[int]
+    properties: InitVar[Dict[str, np.ndarray]] = None
+
+    rotation: float = 0.0
+    translation: Property[np.ndarray, None, np.asarray] = np.asarray(0)
+    anchor: Property[Anchor, str, Anchor] = Anchor.CENTER
+    color: Property[
+        Union[np.ndarray, str], None, transform_single_color
+    ] = 'cyan'
+    size: float = 12.0
+    blending: Property[Blending, str, no_opaque] = Blending.TRANSLUCENT
+    visible: bool = True
+    values: Optional[np.ndarray] = None
+    mode: Property[TextMode, str, TextMode] = TextMode.NONE
+
+    __equality_checks__ = {
+        'values': np.array_equal,
+        'translation': np.array_equal,
+        'color': np.array_equal,
+    }
+
+    def __post_init__(self, text, n_text, properties):
+        self.events.add(text=None)
```

Comment:
Do we need this custom `text` event? Would be nice to drop. People can always connect using `events.connect` if they want to listen to everything

Reply:
I think I see why @kevinyamauchi did this given the current API of TextManager and I did the same in [PR 2885](https://github.com/napari/napari/pull/2885). The key issue is that `values` contains the text values that this class generates, but those are always generated through the `_set_data` method. We might be able to expose that publicly as an attribute in our EventedModel and then use that attribute's associated event, but that might need a few other API changes (which might actually be positive).